avm/res/network/virtual-network-gateway/main.bicep

```diff
@@ -494,6 +494,28 @@ output activeActive bool = virtualNetworkGateway.properties.activeActive
 @description('The location the resource was deployed into.')
 output location string = virtualNetworkGateway.location
 
+@description('The ASN (Autonomous System Number) of the virtual network gateway.')
+output asn int? = isBgp ? virtualNetworkGateway.properties.bgpSettings.asn : null //'Not applicable (No Bgp)'
+
+@description('The public IP address of the virtual network gateway.')
+output publicIpAddress string = !empty(existingFirstPipResourceId) ? reference(existingFirstPipResourceId,'2023-04-01').ipAddress: publicIPAddress[0].outputs.ipAddress
```

Comment:
can't this be retrieved as a virtualNetworkGateway property? No matter if existing or not before the deployment?

Reply:
Also in this case we could provide primary and secondary pip as part of a publicIp output object